packages/node/src/indexer/blockDispatcher/worker-block-dispatcher.service.ts

```diff
@@ -245,6 +264,12 @@ export class WorkerBlockDispatcherService
     void this.queue.put(processBlock);
   }
 
+  private async maxBatchSize(workerIdx: number): Promise<number> {
+    const memLeft = await this.workers[workerIdx].getMemoryLeft();
```

Comment:
Maybe we could use https://nodejs.org/api/worker_threads.html#workerresourcelimits instead here. It would be great to remove the async changes to enqueuBlock

Reply:
this wouldn't give us the amount of heap being used